explorer-api/src/mix_nodes/tasks.rs

```diff
@@ -0,0 +1,26 @@
+use crate::state::ExplorerApiStateContext;
+
+pub(crate) struct MixNodesTasks {
+    state: ExplorerApiStateContext,
+}
+
+impl MixNodesTasks {
+    pub(crate) fn new(state: ExplorerApiStateContext) -> Self {
+        MixNodesTasks { state }
+    }
+
+    pub(crate) fn start(self) {
+        info!("Spawning mix nodes task runner...");
+        tokio::spawn(async move {
+            let mut interval_timer = tokio::time::interval(std::time::Duration::from_secs(60 * 60)); // every hour
+            loop {
+                // wait for the next interval tick
+                interval_timer.tick().await;
+
+                info!("Updating mix node cache...");
+                self.state.inner.mix_nodes.refresh().await;
+                info!("Done");
```

Comment:
This is just a side note, but it looks like a perfect use case for `tracing` - https://tokio.rs/blog/2019-08-tracing. We're not really going to know what this `Done` refers to due to logs coming from different places. But I think `tracing` spans solve that issue. Anyway, just wanted to point out existence of that library (and that's something I wanted to check out myself at some point)

Reply:
Yup, makes sense. At the moment, these logs do not go anywhere. On a system that aggregated them, trace ids would be very useful.